tools/vspec2csv.py

```diff
@@ -81,18 +82,121 @@ def format_data(json_data):
     return f"{Id},{Type},{DataType},{Unit},{Min},{Max},{Desc},{Enum},{Sensor},{Actuator}"
 
 
-def json2csv(json_data, file_out, parent_signal):
-    for k in list(json_data.keys()):
+def json2csv(json_data, file_out, parent_signal, ext=[]):
+    for k in json_data.keys():
         if (len(parent_signal) > 0):
             signal = parent_signal + "." + k
         else:
             signal = k
 
+        local_ext = ext
+        if ('extension' in json_data[k].keys()):
+            local_ext = ext + [(json_data[k]['extension'])]
+
+        # if it's a branch, create an entry and continue
+        # with its children
         if (json_data[k]['type'] == 'branch'):
             file_out.write(signal + ',' + format_data(json_data[k]) + '\n')
-            json2csv(json_data[k]['children'], file_out, signal)
+            # if postion attribute exists, keep it
+            json2csv(json_data[k]['children'], file_out, signal, local_ext)
+
+        # if it's a leave, make an entry, check the extension entries and
+        # create an entry for every extension
         else:
-            file_out.write(signal + ',' + format_data(json_data[k]) + '\n')
+            if not local_ext:
+                file_out.write(signal + ',' + format_data(json_data[k]) + "," + str(ext) + '\n')
+
+            createExtensionEntries(local_ext, file_out, json_data[k], signal)
+
+
+
+def createExtensionEntries(ext, file_out, json_data, prefix=''):
+    """Function creates the extension entries to a given node as
+       postfix of the node name (e.g. node.Row1.LEFT)
+
+    Parameters
+    ----------
+    ext : list
+        list of branches with extension argument in the path
+    file_out : file
+        csv file to write into
+    json_data : json object
+        json encoded information about the node
+    prefix : string
+        node name before the extension information
+    """
+
+    REG_EX = "\w+\[\d+,(\d+)\]"
+
+    if not ext and file_out and json_data:
+        return
+
+    rest = None
+    i = []
+    if len(ext) == 1:
+        i = ext[0]
+    else:
+        i = ext[0]
+        rest = ext[1:]
+
+
+
+    if prefix and not prefix.endswith("."):
+        prefix += "."
+
+
+    # parse string extension elements (e.g. Row[1,5])
+    if isinstance(i,str):
+        if re.match(REG_EX, i):
+            extRangeArr = re.split("\[+|,+|\]",i)
+            for r in range(int(extRangeArr[1]),int(extRangeArr[2])+1):
+                nextPrefix = prefix + extRangeArr[0]+str(r)
+                if rest:
+                    createExtensionEntries(rest, file_out, json_data, nextPrefix)
+                else:
+                    file_out.write(nextPrefix + ',' + format_data(json_data) + "," + '\n')
+
+
+        # TODO: right now dynamic extensions not supported
+        else:
+            raise vspec.VSpecError("","","Extension type not supported")
+    # Use list elements for extension (e.g. [LEFT,RIGHT])
```

Comment:
# Use list elements for extension (e.g. ["LEFT","RIGHT"])

Reply:
>> I'd propose to go in two steps. First let's discuss the specification in #130 and then on the tooling and representation

I think the hard one is the resulting tree, not the YAML, so I think they need to be discussed together, or start with the Tool result. 
Personally I find the "virtual" node model (see my previous comment) a bit problematic, that for example the node Tilt physically exists in YAML, but not in CSV.
I mentioned the scenario with dynamic registration into this virtual node as an example, please comment how that could be handled.

Reply:
Thanks, @rtroncy. Noted to be fixed.

Reply:
Generalize multidimensional instantiation also to list elements.
E.g.
```
extension:
    - Row[1,4]
    - ["Left","Right"]
```
becomes
```
extension: [ Row[1,4], ["Left","Right"]]
```